app/controllers/unattended_controller.rb

```diff
@@ -69,8 +72,51 @@ def preview?
 
   def render_custom_error(status, error_message, params)
     logger.error error_message % params
+    message = _(error_message) % params
+    return render_ipxe_message(message: message, status: status) if ipxe_request?
     # add a comment character (works with Red Hat and Debian systems) to avoid parsing errors
-    render(:plain => '# ' + _(error_message) % params, :status => status, :content_type => 'text/plain')
+    render(:plain => '# ' + message, :status => status, :content_type => 'text/plain')
+  end
+
+  def render_ipxe_template
+    ipxe_template_kind = TemplateKind.find_by(name: 'iPXE')
+
+    if !@host
+      name = ProvisioningTemplate.global_template_name_for('iPXE')
+      template = ProvisioningTemplate.find_global_default_template(name, 'iPXE')
+
+      unless template
+        render_ipxe_message(message: _("Global iPXE template '%s' not found") % name)
+        return
+      end
+    elsif @host && !@host.build?
+      return unless verify_found_host
+
+      name = @host.local_boot_template_name(:ipxe) || ProvisioningTemplate.local_boot_name(:iPXE)
+      template = ProvisioningTemplate.find_by(name: name, template_kind: ipxe_template_kind)
+      template ||= ProvisioningTemplate.new(
+        name: 'iPXE default local boot fallback',
+        template: Foreman::Ipxe::StubRenderer.new(
+          message: 'iPXE default local boot fallback'
+        ).to_s
```

Comment:
Can't we error out simply? What is the idea behind the stub template?

Reply:
This particular code was written before I'd set up a proper local boot template, so it could probably be removed now in favour of the real template instead.